File: android/src/main/java/me/carda/awesome_notifications/awesome_notifications_android_core/enumerators/ForegroundStartMode.java
package me.carda.awesome_notifications.awesome_notifications_android_core.enumerators;

import static android.app.Service.START_NOT_STICKY;
import static android.app.Service.START_REDELIVER_INTENT;
import static android.app.Service.START_STICKY_COMPATIBILITY;
import static android.app.Service.START_STICKY;

public enum ForegroundStartMode {

    stick,
    stickCompatibility,
    notStick,
    deliverIntent;

    public int toAndroidStartMode() {
        switch (this){
            case notStick: return START_NOT_STICKY;
            case stickCompatibility: return START_STICKY_COMPATIBILITY;
            case deliverIntent: return START_REDELIVER_INTENT;
            case stick:
            default:
                return START_STICKY;
        }
    }
}

File: lib/src/enumerators/foreground_start_mode.dart
/// Determines to operational system how it should behaves in case of get terminated
enum ForegroundStartMode {
  /// Corresponds to [`Service.START_STICKY_COMPATIBILITY`](https://developer.android.com/reference/android/app/Service#START_STICKY_COMPATIBILITY).
  stickCompatibility,

  /// Corresponds to [`Service.START_STICKY`](https://developer.android.com/reference/android/app/Service#START_STICKY).
  stick,

  /// Corresponds to [`Service.START_NOT_STICKY`](https://developer.android.com/reference/android/app/Service#START_NOT_STICKY).
  notStick,

  /// Corresponds to [`Service.START_REDELIVER_INTENT`](https://developer.android.com/reference/android/app/Service#START_REDELIVER_INTENT).
  redeliverIntent
}
